examples/Chat/Example06_ChatSerialization.cs

```diff
@@ -0,0 +1,104 @@
+﻿using NUnit.Framework;
+using OpenAI.Chat;
+using System;
+using System.ClientModel.Primitives;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Text.Json;
+
+namespace OpenAI.Examples;
+
+public partial class ChatExamples
+{
+    #region
+    public static List<ChatMessage> DeserializeMessages(BinaryData data)
+    {
+        List<ChatMessage> messages = [];
+        using JsonDocument messagesAsJson = JsonDocument.Parse(data.ToString());
+
+        foreach (JsonElement jsonElement in messagesAsJson.RootElement.EnumerateArray())
+        {
+            BinaryData jsonElementAsData = BinaryData.FromString(jsonElement.ToString());
+            ChatMessage message = ModelReaderWriter.Read<ChatMessage>(jsonElementAsData, ModelReaderWriterOptions.Json);
+            messages.Add(message);
+        }
+
+        return messages;
+    }
+    #endregion
+
+    #region
+    public static BinaryData SerializeMessages(IEnumerable<ChatMessage> messages)
+    {
+        using MemoryStream stream = new();
+        using Utf8JsonWriter writer = new(stream);
+
+        writer.WriteStartArray();
+        foreach (ChatMessage message in messages)
+        {
+            (message as IJsonModel<ChatMessage>).Write(writer, ModelReaderWriterOptions.Json);
+        }
+        writer.WriteEndArray();
+        writer.Flush();
+
+        return BinaryData.FromBytes(stream.ToArray());
+    }
+    #endregion
+
+    [Test]
+    public void Example06_ChatSerialization()
```

Comment:
Likewise to the Embeddings example, it seems like it'd be very valuable to link to this (with mention) from the README!